### PR TITLE
Integrate Jira issue Indexing

### PR DESCRIPTION
**Pull request by hoxhaeris**
The Jira issues are indexed in parallel with the Bugzilla bugs.
Jira is separated from Bugzilla  (different cache, different persistence store).
The search on the UI is as well separated. 

Search:
We use a  JQL query to filter the search (`--jira-search` flag).

UI: 
bug+issue+junit is made the default selection. Other options include:

- bug+issue+junit (new)
- bug+issue (new)
- bug
- issue (new)
- junit
- build-log
- all

At the current state, we save these fields from Jira cards: 

- ID
- Fields.Summary
- Fields.Status
- Fields.Resolution
- Fields.Priority
- Fields.Creator
- Fields.Assignee
- Fields.Labels
- Target_Versions (custom field)

### CODE REVIEW

File: cmd/search/main.go
Comment (bradmwilliams) [inline cmd/search/main.go:37]: Empty comment?

File: cmd/search/main.go
Comment (bradmwilliams) [inline cmd/search/main.go:94]: Copy-n-paste issues (Jira != Bugzilla) and an empty comment.

File: cmd/search/main.go
Comment (bradmwilliams) [inline cmd/search/main.go:129]: Empty comment

File: cmd/search/main.go
Comment (bradmwilliams) [inline cmd/search/main.go:150]: I would group the individual pieces together.  Then you dont have `//jira` comments every couple lines.

File: cmd/search/main.go
Comment (bradmwilliams) [inline cmd/search/main.go:315]: This seems suspect...it's inside a `bugs/` block of code?  Why switch it to `Issue`?

File: cmd/search/main.go
Comment (bradmwilliams) [inline cmd/search/main.go:334]: Same here...
This seems suspect...it's inside a `bugs/` block of code?  Why switch it to `Issue`?

File: cmd/search/main.go
Comment (hoxhaeris) [reply cmd/search/main.go]: Removed

File: cmd/search/main.go
Comment (hoxhaeris) [reply cmd/search/main.go]: Fixed

File: cmd/search/main.go
Comment (hoxhaeris) [reply cmd/search/main.go]: Fixed

File: cmd/search/main.go
Comment (hoxhaeris) [reply cmd/search/main.go]: Done

File: cmd/search/main.go
Comment (hoxhaeris) [reply cmd/search/main.go]: Most likely some IDE refactoring went wrong. Fixed.

File: cmd/search/main.go
Comment (hoxhaeris) [reply cmd/search/main.go]: Most likely some IDE refactoring went wrong. Fixed.

File: cmd/search/http.go
Comment (stbenjam) [inline cmd/search/http.go:122]: Is it possible to do bug+issue+junit as the default? If someone is searching for a problem w/ a test, it could help to have jira issues shown as well as bugs.

Comment (hoxhaeris) [reply cmd/search/http.go]: Done; bug+issue+junit is added to the options list and made the default selection.

**Comment by hoxhaeris** [comment]
/hold
The `OCPBUGS` project is still not defined as public. We need to clarify that before merging.

**Review by bradmwilliams — COMMENTED** [review]
@hoxhaeris Can you squash your commits and we'll shoot to merge in tomorrow AM?

**Review by bradmwilliams — APPROVED** [review]

**Comment by dgoodwin** [comment]
That helps, thanks, we can work with that.

**Comment by bradmwilliams** [comment]
@dgoodwin I have created a new bug that gets picked up by ci-search: https://issues.redhat.com/browse/OCPBUGS-10
It should show up in the local cache on the next refresh cycle.

**Comment by bradmwilliams** [comment]
> Is there any way we could see API responses for jira issues?

I have a couple test harness' that I have been using to interact with the various layers for our Jira Migration work here:
https://github.com/bradmwilliams/jira-migration

**Comment by dgoodwin** [comment]
Is search supposed to pick up text in the ocpbugs description? I see the above test bug with: https://search.ci.openshift.org/?search=Test+Bug&maxAge=48h&context=1&type=issue&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

But not with: https://search.ci.openshift.org/?search=This+is+a+test+bug+intended&maxAge=48h&context=1&type=issue&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

**Comment by dgoodwin** [comment]
Terrific, thank you!

**Pull request merged**
